### DISCUSSION TITLE
Why next/link doesn't works when href is pointer to an existing element with id in the same page?

### DISCUSSION THREAD

**Discussion by diecodev** [vercel/next.js#44604] [Help]
### Example
```jsx
// --- app/components/header.tsx ---------
import Link from 'next/link'

export const header = () => (
  <header>
    <Link href='#about'>About Us</Link>
  </header>
)
```

```jsx
// --- app/components/about.tsx ---------
export const AboutUs = () => (
  <section id="about">
    {// some code}
  </section>
)
```

```jsx
// --- app/page.tsx ---------
import { Header } from './components/header'
import { AboutUs } from './components/about'

export default function Home() {
  return (
    <main id="about">
      <Header />
      <AboutUs />
      {// some code}
    </main>
)}
```

If you try the code aboive, it's not gonna works in the ***latest version of nextjs 13 and app directory***

**Comment by knnedy** [comment]
There is a typo in your link href attribute you should use "/about" instead of "#about".
```
// --- app/components/header.tsx ---------
import Link from 'next/link'

export const header = () => (
  <header>
    <Link href='/about'>About Us</Link>
  </header>
)
```

**Reply by diecodev** [reply]
I appreciate your quick answer, but I don't want to link to /about page, that route doesn't exist. I just want to scroll to the section with the id='about'. if you use my example with normal anchor tag it works. But, whit the nextjs Link component, it doesn't. The werido here is that, in nextjs docs they say it should works [(Click here to see the exact example in docs)](https://nextjs.org/docs/api-reference/next/link#disable-scrolling-to-the-top-of-the-page).

**Comment by icyJoseph** [comment]
Hi,

I think the problem is that your UI is rendering two `about` id's.

If we translate your example to html:

```html
<main id="about">
    <header>
       <Link href='#about'>About Us</Link>
     </header>
     <section id="about">
         <p>content</p>
     </section>
</main>
```

So when the link looks for an element to scroll to, it finds the top level about, which wraps the entire page.

**Reply by diecodev** [reply]
Oh, sorry, that was a writing error, mb. I try the solution you say and the link component does not work still. Idk why this happen, I also open a issue. Thanks to all of you for try to help me.

**Reply by icyJoseph** [reply]
Alright, but then you need to provide a minimal reproduction demo. It might be that there's something inherent to your setup that's not working. Starting an app from zero I can navigate to fragments when I click on anchor tags, no issue.

**Reply by icyJoseph** [reply]
I also found, https://github.com/vercel/next.js/issues/44622, which talks about next/image issues, is that the issue you talk about?

**Reply by icyJoseph** [reply]
When I go to https://diecodev.vercel.app/ and clicking about scrolls to the about section, but you have done that with JavaScript right?

**Reply by icyJoseph** [reply]
And now I am able to reproduce this, very weird :/ I think I need to read the docs a bit.
